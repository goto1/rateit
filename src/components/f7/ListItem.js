import React from "react";
import PropTypes from "prop-types";
import Icon from "./Icon";

const ListItem = ({ icon, url, children }) =>
  <li>
    <a href={url} className="item-link item-content">
      <div className="item-media">
        <Icon material={icon} />
      </div>
      <div className="item-inner">
        {children}
      </div>
    </a>
  </li>;

ListItem.propTypes = {
  icon: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
};

export default ListItem;
