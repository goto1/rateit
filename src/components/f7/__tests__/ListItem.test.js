import React from "react";
import ListItem from "../ListItem";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("ListItem", () => {
  const HelloWorld = "HelloWorld!";

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ListItem icon="home" url="/home">
          {HelloWorld}
        </ListItem>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains title of the item", () => {
    const wrapper = shallow(
      <ListItem icon="home" url="/home">
        {HelloWorld}
      </ListItem>
    );
    expect(wrapper.find(".item-inner")).toHaveText("HelloWorld!");
  });
});
