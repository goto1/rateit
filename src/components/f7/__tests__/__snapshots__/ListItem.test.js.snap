// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<li>
  <a
    className="item-link item-content"
    href="/home"
  >
    <div
      className="item-media"
    >
      <i
        className="material-icons icon"
      >
        home
      </i>
    </div>
    <div
      className="item-inner"
    >
      HelloWorld!
    </div>
  </a>
</li>
`;
